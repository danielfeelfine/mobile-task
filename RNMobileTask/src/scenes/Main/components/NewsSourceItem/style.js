import { StyleSheet } from 'react-native'
import { colors } from '@styles/base'

const styles = StyleSheet.create({
  item: {
    paddingVertical: 8,
    paddingHorizontal: 16,
    backgroundColor: colors.Light,
  },
  name: {
    marginBottom: 4,
    fontSize: 18,
    fontWeight: '800',
    color: colors.Dark,
  },
  description: {
    marginVertical: 2,
    fontWeight: '300',
    color: colors.Dark,
  },
  url: {
    marginVertical: 2,
    fontSize: 12,
    fontWeight: '300',
    color: colors.Gray,
  }
})

export default styles
