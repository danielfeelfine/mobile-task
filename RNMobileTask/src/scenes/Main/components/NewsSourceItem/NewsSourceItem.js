// @flow
import React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import PropTypes from 'prop-types'
import style from './style'

type Props = {
  item: Object,
  onPress: Function
}

const NewsSourceItem = (props: Props) => {
  return (
    <TouchableOpacity style={style.item} onPress={props.onPress}>
      <Text style={style.name}>{props.item.name}</Text>
      <Text style={style.description}>{props.item.description}</Text>
      <Text style={style.url}>{props.item.url}</Text>
    </TouchableOpacity>
  )
}

NewsSourceItem.propTypes = {
  item: PropTypes.shape({
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired,
  onPress: PropTypes.func.isRequired
}

export default NewsSourceItem
