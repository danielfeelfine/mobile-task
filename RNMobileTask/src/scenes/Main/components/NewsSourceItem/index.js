import NewsSourceItem from './NewsSourceItem'
export default NewsSourceItem