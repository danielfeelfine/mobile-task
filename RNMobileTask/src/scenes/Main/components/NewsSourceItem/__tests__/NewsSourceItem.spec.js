import React from 'react'
import { shallow } from 'enzyme'
import NewsSourceItem from './../NewsSourceItem'


describe('NewsSourceItem component', () => {

  it('renders correctly', () => {
    const props = {
      item: {
        name: 'Item name',
        description: 'Item description',
        url: 'https://item.url',
      },
      onPress: () => {}
    }

    const wrapper = shallow(
      <NewsSourceItem item={props.item} onPress={props.onPress} />
    )

    expect(wrapper).toMatchSnapshot()
  })

})
