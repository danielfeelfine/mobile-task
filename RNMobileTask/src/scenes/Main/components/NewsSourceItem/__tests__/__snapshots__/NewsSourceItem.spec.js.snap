// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsSourceItem component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsSourceItem
    item={
      Object {
        "description": "Item description",
        "name": "Item name",
        "url": "https://item.url",
      }
    }
    onPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "fontSize": 18,
              "fontWeight": "800",
              "marginBottom": 4,
            }
          }
        >
          Item name
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "fontWeight": "300",
              "marginVertical": 2,
            }
          }
        >
          Item description
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#999999",
              "fontSize": 12,
              "fontWeight": "300",
              "marginVertical": 2,
            }
          }
        >
          https://item.url
        </Text>,
      ],
      "onPress": [Function],
      "style": Object {
        "backgroundColor": "#FFFFFF",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Item name",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "fontSize": 18,
            "fontWeight": "800",
            "marginBottom": 4,
          },
        },
        "ref": null,
        "rendered": "Item name",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Item description",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#333333",
            "fontWeight": "300",
            "marginVertical": 2,
          },
        },
        "ref": null,
        "rendered": "Item description",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "https://item.url",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#999999",
            "fontSize": 12,
            "fontWeight": "300",
            "marginVertical": 2,
          },
        },
        "ref": null,
        "rendered": "https://item.url",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "fontSize": 18,
                "fontWeight": "800",
                "marginBottom": 4,
              }
            }
          >
            Item name
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#333333",
                "fontWeight": "300",
                "marginVertical": 2,
              }
            }
          >
            Item description
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#999999",
                "fontSize": 12,
                "fontWeight": "300",
                "marginVertical": 2,
              }
            }
          >
            https://item.url
          </Text>,
        ],
        "onPress": [Function],
        "style": Object {
          "backgroundColor": "#FFFFFF",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Item name",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "fontSize": 18,
              "fontWeight": "800",
              "marginBottom": 4,
            },
          },
          "ref": null,
          "rendered": "Item name",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Item description",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#333333",
              "fontWeight": "300",
              "marginVertical": 2,
            },
          },
          "ref": null,
          "rendered": "Item description",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "https://item.url",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#999999",
              "fontSize": 12,
              "fontWeight": "300",
              "marginVertical": 2,
            },
          },
          "ref": null,
          "rendered": "https://item.url",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
